toolbox/deps_update/main.go

```diff
@@ -188,10 +141,7 @@ func updateDependenciesOf(repo string) error {
 	if err := u.SerializeDeps(istioDepsFile, &deps); err != nil {
 		return err
 	}
-	if _, err := u.Shell("git commit -am Update_Dependencies"); err != nil {
-		return err
-	}
-	if _, err := u.Shell("git push --set-upstream origin " + branch); err != nil {
+	if err := u.CreateCommitPushToRemote(branch, "Update_Dependencies"); err != nil {
```

Comment:
Note that there is an API to update files directly. So you could just use that instead of having the create a local repo.